### PR TITLE
Move Ignored properties into IBeatmap

### LINKED ISSUE #20883: Editor does not read properties ignored on `BeatmapInfo` correctly
### Type

Game behaviour

### Bug description

When importing a beatmap, lazer will always default to using `CountdownType.Normal` with 0 offset, ignoring any values that the imported beatmap is using. This both applies to beatmaps imported from stable using the in-game wizard, .osz files and the in-game beatmap browser. 

It seems like changing the offset in the editor actually gets saved, but turning off or changing the countdown speed is not.

~~I'll see if I can fix this myself as I need it for #20584.~~ see comment

### Screenshots or videos

Stable:
image

Lazer:
image

image

### Version

master

### Logs

-

### PR DESCRIPTION
**Pull request by Ryooshuu**
Closes #20883

### CODE REVIEW

**Comment by bdach** [comment]
Context which isn't in the OP but should be, from discord:

- https://discord.com/channels/188630481301012481/188630652340404224/1103261923480453180
- https://discord.com/channels/188630481301012481/188630652340404224/1104423316887961650

File: osu.Game/Beatmaps/IBeatmap.cs
Comment (bdach) [inline osu.Game/Beatmaps/IBeatmap.cs:58]: So while I don't disagree with doing this in the first place (since I'm the one who suggested it), I don't think this goes far enough. At the *least* the same thing should also be done to `CountdownOffset`:

https://github.com/ppy/osu/blob/b4a64701f5baa923d7ce77a692f271a0e55a16a3/osu.Game/Beatmaps/BeatmapInfo.cs#L179-L182

because it's weird that now half of the countdown info is in `IBeatmap` and half of it is in `BeatmapInfo`. But really, the same logic can be extended to [everything in this `#region`](https://github.com/ppy/osu/blob/b4a64701f5baa923d7ce77a692f271a0e55a16a3/osu.Game/Beatmaps/BeatmapInfo.cs#L137-L184).

If there is a time to move that stuff out of there, now is probably one of the best times to do so. @ppy/team-client would appreciate opinions on this matter.

File: osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs
Comment (bdach) [inline osu.Game/Rulesets/Difficulty/DifficultyCalculator.cs:310]: This should be in the `Delegated IBeatmap implementation` region below.

File: osu.Game/Beatmaps/IBeatmap.cs
Comment (Ryooshuu) [reply osu.Game/Beatmaps/IBeatmap.cs:58]: I was afraid to remove it since I didn't want to cause any unnecessary Realm changes. Albeit, rethinking it now, I don't think it would cause many issues. I'll wait for a response from others however (and I'll probably do another pull request just for this if this is the direction we want to go ahead with).

I'm a bit curious however about how I would name the new class if we decide to go in this direction, I'm currently thinking of `BeatmapDetails` but I'm unsure if that'll stick.

Comment (peppy) [reply osu.Game/Beatmaps/IBeatmap.cs:58]: #23432 strangely relevant here.

I agree with moving all of these out as a first step.

> If we do decide to move everything then it may be better to wrap all of that in some object rather than move everything to `IBeatmap`, because `ProgressiveCalculationBeatmap` is going to grow by quite a large margin if moving properties 1:1. But maybe that's fine.

I think if anything it should be a separate interface, rather than a separate class.

**Comment by irisu01** [comment]
it appears as if master got forcefully merged

File: osu.Game/Beatmaps/IBeatmap.cs
Comment (irisu01) [reply osu.Game/Beatmaps/IBeatmap.cs]: > I think if anything it should be a separate interface, rather than a separate class.

I decided to make a proper interface and a class called `IBeatmapSettings` so that it can be easily added to `IBeatmap`, and it condenses all of the options into one proper area without inflating the size of classes such as `ProgressiveCalculationBeatmap` with properties.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs
Comment (peppy) [inline osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:268]: what makes this a setting? as per my comment, i don't think this should be in a separate class.

maybe editorsettings could exist, but this is not that.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs
Comment (irisu01) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:268]: I believe `Settings` is the best term for these properties because they are all values that can be assigned and set by a medium such as the editor to a beatmap. The other only good term would've been `Metadata`, but that's already taken.

Comment (peppy) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:268]: Disagree.

Now there's `IBeatmap.Settings`, `BeatmapInfo.UserSettings` and then everything in between (`ControlPoints` are also "settings" but they aren't in settings --- where do you draw the line).

Strongly against adding a new usage of the raw term "Settings" anywhere liek this.

Comment (bdach) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:268]: Agree that "setting" isn't good, and agree in principle with splitting an interface for this additional stuff. The biggest issue I have though is how to name that new interface (classic...)

Like... best I can think of is probably `IGameplayBeatmapProperties` or something? And that still sucks.

**Comment by bdach** [comment]
Also what is happening with the history here, out of curiosity? There are two github users committing to one branch? How does that work?

**Comment by irisu01** [comment]
There's nothing really stopping two people from collaborating on the same branch, and the pull request will reflect every commit any party creates to it; which is the entire gist to how maintainers can even push commits to branches on forks that aren't theirs.

Here, we're doing it because we're collaboratively working on the same exact issue; and will do so for most of our branches as we aim to share the same goals with furthering the progress of lazer on our own whims.

**Comment by bdach** [comment]
Ok, I thought that was a git misconfiguration or something. Carry on I guess.

File: osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs
Comment (irisu01) [reply osu.Game/Beatmaps/Formats/LegacyBeatmapDecoder.cs:268]: `IBeatmapProperties` is *far* better if we're going to keep the word "Properties".

**Pull request closed without merging**
